### ISSUE TITLE
wp_delete_post_revision() flushes complete cache

### ISSUE THREAD

**Issue by ouun** [pluginkollektiv/cachify#245]
**Describe the bug**
Thanks for the ongoing work on this simple and great plugin.
I just realized that the automatic processing function [`wp_delete_post_revision()`](https://developer.wordpress.org/reference/functions/wp_delete_post_revision/) calls [`wp_delete_post()`](https://developer.wordpress.org/reference/functions/wp_delete_post/) to delete the revisions. This **flushes the complete Cachify cache** as `do_action('before_delete_post')` [is triggered](https://github.com/pluginkollektiv/cachify/blob/b47f58a39daa4cece961706fcf81c2e2b5c4c2ea/inc/class-cachify.php#L1106).

This flushes the complete page cache unnecessarily as post revisions won't get cached at all. Deleting them has no impact on the frontend.

**To Reproduce**
1. Limit revisions in wp-config: `define('WP_POST_REVISIONS', 1)`
1. Edit a post that has already revisions
2. Note that `wp_delete_post_revision()` is executed
3. Check your cache
4. Realize it is gone

**Expected behavior**
Cache flushes only for posts with status "publish". The hook `before_delete_post` can pass `$post_id` and `$post` to the `flush_total_cache()` function. A check for the post status could help, here. Please let me know if you are open for a PR into this direction.

**System (please complete the following information):**
- Version 2.3.2

**Comment by Zodiac1978** [comment]
> Please let me know if you are open for a PR into this direction.

Hi @ouun and thanks for your issue!

I haven't looked myself at the code, but your explanation and solution looks fine for me. Would be great if you submit a PR for further discussions.

**Comment by ouun** [comment]
Hi @Zodiac1978,

thanks for your answer. I just went ahead and dived a bit deeper into Cachify. Got a bit confused as I realized that the `wp_trash_post` hook will clear the complete cache, too. 
Nevertheless there is a method `delete_item()` for individual post cache flushing available for each Caching Method (DB, HDD,...) and `cachify_remove_post_cache` as a hook, that is never executed.
However I started with a PR here and therefore close this one: https://github.com/pluginkollektiv/cachify/pull/247
